ibm_db2.c

```diff
@@ -2354,10 +2354,11 @@ static int _php_db2_connect_helper( INTERNAL_FUNCTION_PARAMETERS, conn_handle **
         /* Check if we already have a connection for this userID & database combination */
         if (isPersistent) {
             zend_resource *entry;
-            hKeyLen = strlen(database) + strlen(uid) + 8;
+            hKeyLen = strlen(database) + strlen(uid) + strlen(password) + 9;
```

Comment:
```suggestion
            hKeyLen = strlen(database) + strlen(uid) + strlen(password) + sizeof("__db2_");
```

Would be a bit more clear what the 9 means.